nominal/nominal.py

```diff
@@ -62,6 +63,61 @@ def get_user() -> User:
     return conn.get_user()
 
 
+def upload_tdms(
+    file: Path | str, name: str | None = None, description: str | None = None, *, wait_until_complete: bool = True
+) -> Dataset:
+    """Create a dataset in the Nominal platform from a tdms file.
+
+    TDMS channel properties must have both a `wf_increment` and `wf_start_time` property to be included in the dataset.
+
+    Channels will be named as f"{group_name}.{channel_name}" with spaces replaced with underscores.
+
+    If `name` is None, the dataset is created with the name of the file with a .csv suffix.
+
+    If `wait_until_complete=True` (the default), this function waits until the dataset has completed ingestion before
+        returning. If you are uploading many datasets, set `wait_until_complete=False` instead and call
+        `wait_until_ingestions_complete()` after uploading all datasets to allow for parallel ingestion.
+    """
+    import numpy as np
+    import pandas as pd
+    from nptdms import TdmsChannel, TdmsFile, TdmsGroup
+
+    path = Path(file)
+    with TdmsFile.open(path) as tdms_file:
+        # identify channels to extract
+        channels_to_export: OrderedDict[str, TdmsChannel] = OrderedDict()
+        group: TdmsGroup
+        for group in tdms_file.groups():
+            channel: TdmsChannel
+            for channel in group.channels():
+                # some channels may not have the required properties to construct a time track
+                if ("wf_increment" in channel.properties) and ("wf_start_time" in channel.properties):
+                    channel_name = f"{channel.group_name.replace(' ', '_')}.{channel.name.replace(' ', '_')}"
+                    channels_to_export[channel_name] = channel
+
+        df = pd.DataFrame.from_dict(
+            {
+                channel_name: pd.Series(data=channel[:], index=channel.time_track(absolute_time=True, accuracy="ns"))
+                for channel_name, channel in channels_to_export.items()
+            }
+        )
+
+        # format for nominal upload
+        time_column = "time_ns"
+        df.index = df.index.set_names(time_column, level=None, inplace=False)
+        df = df.reset_index(inplace=False)
```

Comment:
nit but `inplace=False` is the default, so unnecessary to specify

Reply:
will remove